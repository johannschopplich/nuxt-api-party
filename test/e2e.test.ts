import { fileURLToPath } from 'node:url'
import destr from 'destr'
import { describe, expect, it } from 'vitest'
import { $fetch, setup } from '@nuxt/test-utils'

describe('nuxt-api-party', async () => {
  await setup({
    server: true,
    rootDir: fileURLToPath(new URL('./fixture', import.meta.url)),
  })

  it('fetches data with $testApi', async () => {
    const html = await $fetch('/$testApi')
    expect(getTestResult(html)).toMatchSnapshot()
  })

  it('throws error for invalid response with $testApi', async () => {
    const html = await $fetch('/$testApi-error')
    expect(getTestResult(html)).toMatchSnapshot()
  })

  it('fetches data with useTestApiData', async () => {
    const html = await $fetch('/useTestApiData')
    expect(getTestResult(html)).toMatchSnapshot()
  })
})

function getTestResult(html: string) {
  const content = html.match(/<script\s+type="text\/test-result">(.*?)<\/script>/s)?.[1]
  return destr(content)
}
