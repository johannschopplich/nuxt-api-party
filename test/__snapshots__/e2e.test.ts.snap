// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nuxt-api-party > fetches data with $testApi 1`] = `
{
  "blob": "Foo",
  "json": [
    {
      "completed": false,
      "id": 1,
      "title": "delectus aut autem",
      "userId": 1,
    },
    {
      "completed": false,
      "id": 2,
      "title": "quis ut nam facilis et officia qui",
      "userId": 1,
    },
    {
      "completed": false,
      "id": 3,
      "title": "fugiat veniam minus",
      "userId": 1,
    },
  ],
}
`;

exports[`nuxt-api-party > fetches data with useTestApiData 1`] = `
[
  {
    "completed": false,
    "id": 1,
    "isTransformed": true,
    "title": "delectus aut autem",
    "userId": 1,
  },
  {
    "completed": false,
    "id": 2,
    "isTransformed": true,
    "title": "quis ut nam facilis et officia qui",
    "userId": 1,
  },
  {
    "completed": false,
    "id": 3,
    "isTransformed": true,
    "title": "fugiat veniam minus",
    "userId": 1,
  },
]
`;

exports[`nuxt-api-party > throws error for invalid response with $testApi 1`] = `
{
  "data": {
    "reason": "anything",
  },
  "message": "Not Found",
  "stack": "",
  "statusCode": 404,
  "statusMessage": "Not Found",
  "url": "/api/foo/bar",
}
`;
